src/components/Ballot.tsx

```diff
@@ -10,32 +10,35 @@ import HelpPage from '../pages/HelpPage'
 import NotFoundPage from '../pages/NotFoundPage'
 import PreReviewPage from '../pages/PreReviewPage'
 import PrintPage from '../pages/PrintPage'
-import ResetPage from '../pages/ResetPage'
 import ReviewPage from '../pages/ReviewPage'
 import SettingsPage from '../pages/SettingsPage'
 import StartPage from '../pages/StartPage'
 
 const Ballot = () => {
-  const { election, contests } = useContext(BallotContext)
-
-  const { bmdConfig } = election!
-  const { requireActivation, showHelpPage, showSettingsPage } = bmdConfig!
+  const { ballotStyleId, contests, election, precinctId } = useContext(
+    BallotContext
+  )
+  const {
+    requireActivation,
+    showHelpPage,
+    showSettingsPage,
+  } = election!.bmdConfig!
 
   return (
     <Switch>
-      {requireActivation ? (
-        <Route exact path="/" component={ActivationPage} />
-      ) : (
+      {!requireActivation || (ballotStyleId && precinctId) ? (
         <Redirect exact path="/" to="/start" />
+      ) : (
+        <Route exact path="/" component={ActivationPage} />
       )}
-      <Route exact path="/reset" component={ResetPage} />
       <Route exact path="/activate" component={ActivationPage} />
       <Route path="/cast" component={CastBallotPage} />
       <Route path="/start" exact component={StartPage} />
-      {contests.length && (
-        <Redirect exact from="/contests" to={`/contests/0`} />
-      )}
-
+      <Redirect
```

Comment:
I don't like the style of changing the redirect based on a conditional. Conditional before the action is clearer.

Reply:
I'm not sure I understand what you mean. Isn't this what the code is doing…
```
{contests.length ? '/contests/0' : '/'}
       ^                 ^          ^
  conditional          action     action
```

Reply:
I'm not a fan of having the conditional be on the target of the redirect, when the outcome is so different, but it's also a nit. I should have been clearer.